go/connection_test.go

```diff
@@ -287,6 +287,140 @@ func TestInterpolateParamsUint64(t *testing.T) {
+			expected:    []*string{},
+		},
+		{
+			name:        "Bool",
+			inputArgs:   []driver.Value{true, false},
+			expectedErr: nil,
+			expected:    []*string{aws.String("1"), aws.String("0")},
+		},
+		{
+			name:        "Zero-value time",
+			inputArgs:   []driver.Value{time.Time{}},
+			expectedErr: nil,
+			expected:    []*string{aws.String("'0000-00-00'")},
+		},
+		{
+			name:        "501 nanoseconds is still < 1 microsecond", // From TestConnection_InterpolateParams_Bool
+			inputArgs:   []driver.Value{time.Time{}.Add(1 * time.Nanosecond)},
+			expectedErr: nil,
+			expected:    []*string{aws.String("'0001-01-01 00:00:00'")},
+		},
+		{
+			name:        "For non-zero-value time.Times, Date and time are present, even if time is zero-value",
+			inputArgs:   []driver.Value{testTime},
+			expectedErr: nil,
+			expected:    []*string{aws.String("'2024-07-01 00:00:00'")},
+		},
+		{
+			name:        "Datetime with Microseconds",
+			inputArgs:   []driver.Value{testTimeMicro},
+			expectedErr: nil,
+			expected:    []*string{aws.String("'2024-07-02 01:02:03.123456'")},
+		},
+		{
+			name:        "Byte Slice - Caller must use utils.go/FormatBytes before passing in query args",
+			inputArgs:   []driver.Value{[]byte{'0'}},
+			expectedErr: nil,
+			expected:    []*string{aws.String("0")}, // No change
+		},
+		{
+			name:        "Byte Slice - After FormatBytes",
+			inputArgs:   []driver.Value{FormatBytes([]byte{'0'})},
+			expectedErr: nil,
+			expected:    []*string{aws.String("_binary'0'")},
+		},
+		{
+			name:        "String - Caller must use utils.go/FormatString before passing in query args",
+			inputArgs:   []driver.Value{"This is a string"},
+			expectedErr: nil,
+			expected:    []*string{aws.String("This is a string")}, // No change
+		},
+		{
+			name:        "String - After FormatString",
+			inputArgs:   []driver.Value{FormatString("This is a string with ' single quotes and \n chars")},
+			expectedErr: nil,
+			expected:    []*string{aws.String("'This is a string with '' single quotes and \\n chars'")},
+		},
+		{
+			name:        "Nil -> NULL",
+			inputArgs:   []driver.Value{nil},
+			expectedErr: nil,
+			expected:    []*string{aws.String("NULL")},
+		},
+		{
+			name: "Every supported type",
+			inputArgs: []driver.Value{int64(-10), uint64(42), 1.23, true, testTime, []byte("This is a slice of bytes"),
+				"This is a string"},
+			expectedErr: nil,
+			expected: []*string{aws.String("-10"), aws.String("42"), aws.String("1.23"), aws.String("1"),
+				aws.String("'2024-07-01 00:00:00'"), aws.String("This is a slice of bytes"), aws.String("This is a string")},
+		},
+	}
+	c := createTestConnection(t)
+	for _, tc := range testCases {
+		// https://go.dev/blog/loopvar-preview
+		// Pre-Go-1.22, the loop variable `tc` is shared between each loop iteration. Because t.Parallel() is true,
+		// the test cases run concurrently, and `tc` is likely to change mid-test. We can avoid that by creating a
+		// local copy.
+		tcCopy := tc
```

Comment:
I think idiomatically the accepted approach here is to redefine tc as `tc := tc`, just to avoid usage of original pointer reference by accident

Reply:
Ah that makes sense. I'll make that change.